parser/lib/jsonapi/parser.rb

```diff
@@ -17,14 +17,13 @@ module JSONAPI
 
   # Parse a JSON API document.
   #
-  # @param document [Hash, String] the JSON API document.
+  # @param document [Hash] the JSON API document.
   # @param options [Hash] options
   #   @option options [Boolean] :id_optional (false) Whether the resource
   #     objects in the primary data must have an id.
   # @return [JSONAPI::Parser::Document]
   def parse(document, options = {})
-    hash = document.is_a?(Hash) ? document : JSON.parse(document)
-
-    Parser::Document.new(hash, options)
+    raise Parser::InvalidDocument, 'document must be a hash' unless document.is_a?(Hash)
+    Parser::Document.new(document, options)
```

Comment:
Right, then you can also remove the dependency to `json` in both this file and the gemspec.